sale_invoice_policy/models/product_template.py

```diff
@@ -1,54 +1,29 @@
 # Copyright 2017 ACSONE SA/NV (<http://acsone.eu>)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import api, fields, models
+from odoo import api, models
 
 
 class ProductTemplate(models.Model):
     _inherit = "product.template"
 
-    default_invoice_policy = fields.Selection(
-        [("order", "Ordered quantities"), ("delivery", "Delivered quantities")],
-        string="Default Invoicing Policy",
-        help="Ordered Quantity: Invoice based on the quantity the customer "
-        "ordered.\n"
-        "Delivered Quantity: Invoiced based on the quantity the vendor "
-        "delivered (time or deliveries).",
-        default=lambda self: self.env["ir.default"].get(
-            "res.config.settings", "default_invoice_policy"
-        ),
-    )
-
-    invoice_policy = fields.Selection(
-        compute="_compute_invoice_policy",
-        store=False,
-        readonly=True,
-        search="_search_invoice_policy",
-        inverse="_inverse_invoice_policy",
-        precompute=False,
-    )
-
-    def _inverse_invoice_policy(self):
-        for template in self.filtered("invoice_policy"):
-            template.default_invoice_policy = template.invoice_policy
-
-    @api.depends("detailed_type", "default_invoice_policy")
-    @api.depends_context("invoice_policy")
+    @api.depends("detailed_type")
     def _compute_invoice_policy(self):
```

Comment:
Why still this method if the field is not computed? And IMO, this should be adjusted on the sales order fields, not here.

Reply:
Hi Pedro,

Yes, the field is computed, here is Odoo's declaration of the field:
```
    invoice_policy = fields.Selection(
        [('order', 'Ordered quantities'),
         ('delivery', 'Delivered quantities')], string='Invoicing Policy',
        compute='_compute_invoice_policy', store=True, readonly=False, precompute=True,
        help='Ordered Quantity: Invoice quantities ordered by the customer.\n'
             'Delivered Quantity: Invoice quantities delivered to the customer.')
```